rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -76,18 +76,30 @@ class SeedingRelationApiDualWriteHandler(BaseRelationApiDualWriteHandler):
     _public_tenant: Optional[Tenant] = None
     _platform_default_policy_uuid: Optional[str] = None
     _admin_default_policy_uuid: Optional[str] = None
+    role: Optional[Role] = None
 
-    def prepare_for_update(self, role: Role):
+    def __init__(self, role: Optional[Role] = None, replicator: Optional[RelationReplicator] = None):
+        """Initialize SeedingRelationApiDualWriteHandler."""
+        super().__init__(replicator)
+        self.role = role
+
+    def prepare_for_update(self, role: Optional[Role] = None):
         """Generate & store role's current relations."""
         if not self.replication_enabled():
             return
+        if not role:
+            role = self.role
+        assert isinstance(role, Role)
         self._current_role_relations = self._generate_relations_for_role(role)
 
-    def replicate_update_system_role(self, role: Role):
+    def replicate_update_system_role(self, role: Optional[Role] = None):
```

Comment:
Taking a step back – looks like we have 3 ways now to assign the role. Are these all equivalent though? What is the correct way to use this class? It's not clear to me here and I suspect may be confusing for others.

Reply:
Right, it is better to be consistent, updated the seeding handler to be consistent